### PR TITLE
Added a representation that includes an organization and user model

### LINKED ISSUE #34013: Add More Info to Organization Events
### Description

Currently, it's not possible to determine which user was removed or added to an organization. Therefore, It would be helpful to further add that info to the payload of an organization event.

### Discussion

https://github.com/keycloak/keycloak/issues/31421
https://github.com/keycloak/keycloak/issues/30180

### Motivation

The information can be published to internal APIs

### Details

_No response_

### PR DESCRIPTION
**Pull request by Degoya-Studio**
Added a representation that includes an organization and user model, as well as included it in the event body for removing and adding of members to an organization 

Closes #34013

### CODE REVIEW

**Comment by sguilhen** [comment]
@pedroigor can you run the CI for this one too?

**Review by sguilhen — COMMENTED** [review]
In general I think this approach is ok, I'm just not sure about the event using the full user rep if we can use the brief rep. Is there any info that would be important to display that is not available in the brief user representation?

@pedroigor WDYT?

**Comment by pedroigor** [comment]
@sguilhen @Degoya-Studio Sorry for the delay.

The problem I see here is that admin events are usually related to a specific domain object (e.g.: user, org, etc) not really associations like we are doing here. The way we fire admin events for groups when managing memberships is also the same used by organization memberhips here and we should probably fix that there too. Perhaps as a follow-up. If we agree about having a specific representation, I'm also not sure if we should make it available from `keycloak-core`.

I would say we should have a `details` field in admin events similar to what we have for regular user events so that we can create admin events that are not directly related to managing a specific domain object but a specific operation that involves multiple ones (or multiple representations).

I'm also wondering how we will filter, in the future, events for a specific organization.

We could take the opportunity and fire `ProviderEvent` for organization membership events. I can contribute to this one.

**Comment by Degoya-Studio** [comment]
@pedroigor I'm okay with just providing the member id. If you want I can add a details Hashmap field to the event instead where I provide the id

**Comment by pedroigor** [comment]
>@pedroigor I'm okay with just providing the member id. If you want I can add a details Hashmap field to the event instead where I provide the id

Works for me, but I will ask @mposolda and @stianst if there is a reason (e.g.: representation alone were enough for admin events so far) for not having the `details` map as we have for user events.

**Comment by pedroigor** [comment]
@Degoya-Studio We'll need the `details` map. If you cannot do it (it will require a bit more changes like updating liquibase changesets), please let me know and I'll send a commit to your branch.

**Comment by Degoya-Studio** [comment]
@pedroigor I think it would be better if you take over since I don't know where I need to make these changes. Thank you

**Comment by pedroigor** [comment]
@Degoya-Studio Can you please review my commits? Basically, they are adding both `username` and `email` of the member to the newly added `details` field of the admin events.

File: services/src/main/java/org/keycloak/services/resources/admin/AdminEventBuilder.java [hunk 1]
```diff
@@ -263,6 +265,20 @@ public AdminEventBuilder representation(Object value) {
         return this;
     }
 
+    public AdminEventBuilder detail(String key, String value) {
```
Comment (Degoya-Studio) [inline services/src/main/java/org/keycloak/services/resources/admin/AdminEventBuilder.java:268]: nice builder technique!

File: model/jpa/src/main/java/org/keycloak/organization/jpa/JpaOrganizationProvider.java [hunk 2]
```diff
@@ -191,6 +191,7 @@ private boolean addMember(OrganizationModel organization, UserModel user, Member
             }
 
             user.joinGroup(group, metadata);
+            OrganizationModel.OrganizationMemberJoinEvent.fire(organization, user, session);
```
Comment (Degoya-Studio) [inline model/jpa/src/main/java/org/keycloak/organization/jpa/JpaOrganizationProvider.java:194]: what exactly is the purpose of having this OrganizationMemberJoinEvent interface and firing it here?

**Review by Degoya-Studio — COMMENTED** [review]
Looks great overall, thank you. The only thing I find difficult to understand is how the testing setup works in keycloak.

**Review by sguilhen — APPROVED** [review]

**Review by pedroigor — APPROVED** [review]

**Review by sguilhen — APPROVED** [review]

**Review by ssilvert — APPROVED** [review]
UI part LGTM

**Pull request merged**